### PR TITLE
TreeView -- Inhebit execution of actions for row click.

### PR DESCRIPTION
**Pull request by codingforfun**
In some cases it's usefull to forbit opening FormView or execute `tree_but_open` action for click onto a row in a Tree- or ListView. 
This module provides a new configuration attribute `open` similiar to `create`, `edit` and `delete`.

### CODE REVIEW

**Comment by hbrunn** [comment]
if you have an action defined without form view, the result is the same. What's your use case that you can't simply do this?

**Comment by codingforfun** [comment]
Unfortunately this doesn't apply in general. It does not work if the list view is embedded into a form view. Also I think it's more obvious and to read and easier to define.

File: web_tree_no_open/README.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Web Tree No Open
```
Comment (elicoidal) [inline web_tree_no_open/README.rst:6]: Hi @codingforfun thanks for your contribution.
Please add a line before the document title.

File: web_tree_no_open/README.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+================
+
+This module extends the functionality of tree views.
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: What about?
This module extends the functionality of tree views and adds a new attribute to deactivate the onclick behaviour in them.
The new configuration attribute 'open' is to be used in the same way as existing ones 'edit', 'create' and 'delete'.

File: web_tree_no_open/README.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+The new attribute makes it possible to deactivate the onclick behaviour in tree
+views.
+::
+
+   <tree open="false" >
+
+Installation
+============
+
+To install this module, just follow basic steps to install an odoo module.
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: s/odoo/Odoo

**Comment by ivantodorovich** [comment]
Have you considered also making a PR to odoo?

File: web_tree_no_open/README.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Configuration
+=============
+
+Not needed.
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: No configuration needed.

File: web_tree_no_open/README.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+To use this module, you need to:
+
+* go to ...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/web/8.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: remove

File: web_tree_no_open/README.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+.. branch is "8.0" for example
+
+For further information, please visit:
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: remove

File: web_tree_no_open/README.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: remove if empty

File: web_tree_no_open/README.rst (hunk 1 above)
Comment (elicoidal) [inline web_tree_no_open/README.rst:6]: Not clickable tree view?

File: web_tree_no_open/README.rst [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline web_tree_no_open/README.rst]: https

File: web_tree_no_open/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+{
+    'name': 'web_tree_no_open',
+    'summary': 'Disable actions for click onto Tree- or ListView rows.',
+    'version': '8.0.0.1.0',
```
Comment (elicoidal) [inline web_tree_no_open/__openerp__.py]: 8.0.1.0.0

File: web_tree_no_open/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    'author': 'initOS GmbH, Odoo Community Association (OCA)',
+    'installable': True,
+    'license': 'AGPL-3',
+    'application': False,
+    'installable': True,
+    'auto_install': False,
+    'depends': [
+        'base',
```
Comment (elicoidal) [inline web_tree_no_open/__openerp__.py]: not sure base is necessary

**Comment by codingforfun** [comment]
@hbrunn 

> What's your use case that you can't simply do this?

My use case is that I use a delegate model as a kind of proxy to unify relationships between some objects and some other objects. To list the relationship of an Object to it's connected other objects I list entities of the delegate model. There is a button to open the target of the delegate model. Therefor I do not want the default form view of the delegate model to get opened when it is clicked, because it is of no value to the user. 

My first try was to solve the delegation problem via `tree_but_open` action, but unfortunately this doesn't work on TreeViews embedded into FormViews and is not straight forward to extend.

**Comment by codingforfun** [comment]
@ivantodorovich 

> Have you considered also making a PR to odoo?

You mean as a core patch? No not yet.

**Comment by codingforfun** [comment]
@hbrunn 
Did I answered your question or is there something missing?

**Comment by hbrunn** [comment]
@codingforfun nothing missing, but I'm still in doubt if your approach is the best solution. For your specific problem, I'd create a custom widget that does whatever your button does in `row_clicked`, this should do for a more consistent user experience.
And for the general case, I'll leave it to other reviewers if they think we need this module, I'll abstain.

**Comment by codingforfun** [comment]
@hbrunn 
I do not insist on my solution in detail. If there is a better way it will be fine, however I think the need to create list views that do not react on click is there. If so it should be consistent for stand alone views and views embedded into form views. I think providing another attribute similar to 'edit' ... etc. is a natural approach to provide this. 

You are right. I think sooner or later for my special case I'll take the time and build a custom solution for my use case, but since I had this extension already there I would like to share. 

For my special case the most general solution would be to generalize the list view behaviour to also enable the `tree_but_open` mechanism to work for list views embedded into form views. I already had a look into this, but this looks a bit more complicated.

**Comment by codingforfun** [comment]
Are there more opinions here about this?

**Comment by codingforfun** [comment]
@hbrunn 
You are right in that this probably doesn't make much sense for a list view with associated form view. 
This only makes sense for list views embedded into form view via one2many. 
You think it would be a better PR if I restrict the change to `instance.web.form.One2ManyListView` and `instance.web.form.Many2ManyListView`? 

This is also related to https://github.com/OCA/web/pull/326. 
The same could be achieved by using https://github.com/OCA/web/pull/326 and providing a noop tree_but_open action, although this wouldn't be very intuitive to program.

**Comment by OSguard** [comment]
@codingforfun is this opsolete to https://github.com/OCA/web/pull/326 ?

**Comment by codingforfun** [comment]
> is this opsolete to #326 ?

not really. You can achieve similar effect with #326 by using a noop server action, but in my opinion this is a hacky approach. Doing so has following disadvantages over compared to the approach here: 
- Fair amount of extra code to write
- Not very obvious approach for a clearly defined simple requirement
- Extra RPC calls to get the noop action from the server. 
- The solution here integrates with the other tree options. E.g. you could make it dependent on context or other criteria.

**Comment by OSguard** [comment]
:+1:

**Comment by tremlin** [comment]
Code review :+1:

File: web_tree_no_open/README.rst [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+Configuration
+=============
+
+No configuration needed.
+
+Usage
+=====
+
+To use this module, you need to:
```
Comment (pedrobaeza) [inline web_tree_no_open/README.rst]: Remove this empty paragraph

File: web_tree_no_open/README.rst [hunk 12]
```diff
@@ -0,0 +1,67 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
```
Comment (pedrobaeza) [inline web_tree_no_open/README.rst:3]: Put target link

**Comment by odooCroatia** [comment]
:+1:

**Review by remihb — APPROVED** [review]

**Comment by codingforfun** [comment]
Someone had the same idea here: 
https://apps.odoo.com/apps/modules/10.0/web_tree_no_open/
https://github.com/aydi-nebil/web/tree/12.0/web_tree_no_open
so obviously there is a need for this

**Comment by codingforfun** [comment]
In fact. Actually comparing the code for v10, I think it’s just a copy of mine.

**Pull request closed without merging**